core/chains/evm/txm/storage/inmemory_store_manager.go

```diff
@@ -35,14 +36,14 @@ func (m *InMemoryStoreManager) AbandonPendingTransactions(_ context.Context, fro
 	return fmt.Errorf(StoreNotFoundForAddress, fromAddress)
 }
 
-func (m *InMemoryStoreManager) Add(addresses ...common.Address) error {
+func (m *InMemoryStoreManager) Add(addresses ...common.Address) (err error) {
 	for _, address := range addresses {
 		if _, exists := m.InMemoryStoreMap[address]; exists {
-			return fmt.Errorf("address %v already exists in store manager", address)
```

Comment:
why is it fine to continue updating m.InMemoryStoreMap[address]  in case exists is true? (Before this PR, we would have returned)